app/lib/rocketchat.js

```diff
@@ -1067,14 +1071,30 @@ const RocketChat = {
 	},
 	post(...args) {
 		return new Promise(async(resolve, reject) => {
+			const isMethodCall = args[0]?.startsWith('method.call/');
 			try {
 				const result = await this.sdk.post(...args);
+
+				/**
+				 * if API_Use_REST_For_DDP_Calls is enabled and it's a method call,
+				 * responses have a different object structure
+				 */
+				if (isMethodCall) {
+					const response = JSON.parse(result.message);
+					if (response?.error) {
+						throw response.error;
```

Comment:
Because the idea is to catch the error on the same `try/catch` and evaluate if it's related to TOTP.
If it is, we handle it and ask for TOTP code.
Otherwise then we reject it as you said.